Firestore/core/src/firebase/firestore/core/relation_filter.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_CORE_RELATION_FILTER_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_CORE_RELATION_FILTER_H_
+
+#include <string>
+
+#include "Firestore/core/src/firebase/firestore/core/filter.h"
+#include "Firestore/core/src/firebase/firestore/model/document.h"
+#include "Firestore/core/src/firebase/firestore/model/field_path.h"
+#include "Firestore/core/src/firebase/firestore/model/field_value.h"
+
+namespace firebase {
+namespace firestore {
+namespace core {
+
+/** Represents a filter to be applied to the query. */
+class RelationFilter : public Filter {
+ public:
+  /**
+   * Creates a new filter that compares fields and values. Only intended to be
+   * called from Filter::Create().
+   */
+  RelationFilter(model::FieldPath field, Operator op, model::FieldValue value);
+
+  const model::FieldPath& field() const override;
+
+  bool Matches(const model::Document& doc) const override;
+
+  std::string canonical_id() const override;
```

Comment:
Nit: should we attempt to group properties (field and canonical_id) in this case?

However, this isn't "cheap" in the way that properties normally should be. An equivalent operation in our code is `ToString()` which we don't spell `to_string()`. Perhaps this should be `CanonicalId()`?

Reply:
Done.